inspirehep/modules/orcid/api.py

```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2018 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""APIs for ORCID."""
+
+from __future__ import absolute_import, division, print_function
+
+import requests
+from flask import current_app as app
+from inspire_utils.record import get_value
+from orcid import MemberAPI
+
+from inspirehep.modules.orcid import OrcidConverter
+from inspirehep.modules.orcid.utils import LOGGER, \
+    _get_api_url_for_recid, _split_lists, WORKS_BULK_QUERY_LIMIT, \
+    RECID_FROM_INSPIRE_URL
+
+
+def push_record_with_orcid(recid, orcid, oauth_token, put_code=None):
+    """Push record to ORCID with a specific ORCID ID.
+
+    Args:
+        recid (string): HEP record to push
+        orcid (string): ORCID identifier to push onto
+        oauth_token (string): ORCID user OAUTH token
+        put_code (Union[string, NoneType]): put-code to push record onto,
+            if None will push as a new record
+
+    Returns:
+        string: the put-code of the updated/inserted item
+    """
+    server_name = app.config["SERVER_NAME"]
```

Comment:
If I understand correctly SERVER_NAME is a string like 'http://labs.inspirehep.net', right?
The name is very confusing, but I know this PR did not introduce this bad name.

Reply:
It comes from flask itself.

Reply:
It's a core configuration variable in Flask: http://flask.pocoo.org/docs/0.12/config/

Reply:
Ah ok, then I love it!